### PR TITLE
Add support for pyright language server

### PR DESCRIPTION
**Pull request by mjlbach**
This should add pyright support to lsp-mode with an opt-in +pyright flag.

### CODE REVIEW

File: modules/lang/python/config.el [hunk 1]
```diff
@@ -292,7 +292,7 @@ called.")
 
 
 (use-package! lsp-python-ms
-  :when (and (featurep! +lsp) (not (featurep! :tools lsp +eglot)))
+  :when (and (featurep! +lsp) (not (featurep! :tools lsp +eglot)) (not (featurep! +pyright)))
```
Comment (hlissner) [inline modules/lang/python/config.el:295]: Please make these separate `:when` statements.

```elisp
(use-package! lsp-python-ms
  :when (featurep! +lsp)
  :when (not (featurep! +pyright))
  :when (not (featurep! :tools lsp +eglot))
```

File: modules/lang/python/config.el [hunk 2]
```diff
@@ -308,6 +308,10 @@ called.")
     (ignore-errors (apply orig-fn args))))
 
 
+(use-package!  lsp-pyright  
+  :when (and (featurep! +lsp) (featurep! +pyright) (not (featurep! :tools lsp +eglot)))
```
Comment (hlissner) [inline modules/lang/python/config.el:312]: Please make these separate `:when` statements.

```elisp
(use-package! lsp-pyright 
  :when (featurep! +lsp)
  :when (featurep! +pyright)
  :when (not (featurep! :tools lsp +eglot))
```

File: modules/lang/python/config.el [hunk 3]
```diff
@@ -308,6 +308,10 @@ called.")
     (ignore-errors (apply orig-fn args))))
 
 
+(use-package!  lsp-pyright
```
Comment (hlissner) [inline modules/lang/python/config.el:311]: Please collapse the double space before `lsp-pyright` into a single space.

File: modules/lang/python/config.el [hunk 4]
```diff
@@ -308,6 +308,10 @@ called.")
     (ignore-errors (apply orig-fn args))))
 
 
+(use-package!  lsp-pyright  
+  :when (and (featurep! +lsp) (featurep! +pyright) (not (featurep! :tools lsp +eglot)))
+  :after lsp-clients)
+
```
Comment (hlissner) [inline modules/lang/python/config.el:314]: Please add another blank newline after this use-package! block.

**Comment by mjlbach** [comment]
Thanks! Made the requested changes (I think)

**Comment by mjlbach** [comment]
Note there's still some issues with completions in modules that may be related to the typestubs distributed with pyright, so people might want to defer to using the old language server for now!

**Comment by mjlbach** [comment]
Ok, if you're having issues getting typing, you just need to create stubs in your project directory

```
cd path/to/my/project
source /path/to/my/virtualenv/bin/activate
pyright --createstub tensorflow
```

**Comment by hlissner** [comment]
Understood. Thanks for the PR!

**Comment by jtpavlock** [comment]
I think I'm a bit confused. Is this a language server? 

https://github.com/microsoft/pyright
> Pyright only provides syntax and type checking

**Comment by mjlbach** [comment]
@jtpavlock pylance, the new language server that powers vscode python completion uses is a proprietary wrapper (with an accompanying non-permissive license) around pyright. pyright has a language-server built-in as well (but with some fancy AI-assisted intellisense)

**Comment by jtpavlock** [comment]
> @jtpavlock pylance, the new language server that powers vscode python completion is a proprietary wrapper (with an accompanying non-permissive license) around pyright. Pyright has a language-server built-in as well (but without the fancy AI-assisted intellisense).
> 
> See: [microsoft/pyright#99](https://github.com/microsoft/pyright/issues/99)

Thanks for the reply, I figured that out shortly after posting and tried to save my embarrassment 😄

**Comment by mjlbach** [comment]
No embarrassment at all! So many language servers...

**Pull request merged**